chats/views.py

```diff
@@ -39,55 +43,79 @@ def index(request):
+        fname = request.POST['first_name']
+        lname = request.POST['last_name']
         email = request.POST['email']
-        pass1 = request.POST['pass1']
-        pass2 = request.POST['pass2']
+        pass1 = request.POST['password1']
+        pass2 = request.POST['password2']
 
         if User.objects.filter(username=username):
-            messages.error(request, "Username already exist! Please try other username.")
-            return redirect('register')
+            #messages.error(request, "Username already exist! Please try other username.")
+            return JsonResponse({"status" : "unsuccessfull",
+                                "error": "Username already exist! Please try other username."})
     
         if User.objects.filter(email=email).exists():
-            messages.error(request, "Email is already registered!")
-            return redirect('register')
+            #messages.error(request, "Email is already registered!")
+            return JsonResponse({"status" : "unsuccessfull",
+                                "error": "Email is already registered!"})
+        
+        try:
+            validate_email(email)
+        except:
+            return JsonResponse({"status" : "unsuccessfull",
+                                "error": "Please enter valid email!"})
         
         if len(username) > 20:
-            messages.error(request, "Username must be under 20 characters!")
-            return redirect('register')
+            #messages.error(request, "Username must be under 20 characters!")
+            return JsonResponse({"status" : "unsuccessfull",
+                                "error": "Username must be under 20 characters!"})
         
         if pass1 != pass2:
-            messages.error(request, "Passwords didn't matched!")
-            return redirect('register')
+            #messages.error(request, "Passwords didn't matched!")
+            return JsonResponse({"status" : "unsuccessfull",
+                                "error": "Passwords didn't matched!"})
+
+        passReg = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!#%*?&]{8,}$"
+        pat = re.compile(passReg)
+        if not re.search(pat, pass1):
+            return JsonResponse({"status" : "unsuccessfull",
+                                "error": "Your password must contain at least one small letter, one capital letter, one number, one special symbol and must be at least 8 symbols long."})
 
         myusr = User.objects.create_user(username, email, pass1)
         myusr.first_name = fname
         myusr.last_name = lname
         myusr.is_active = True
         myusr.save()
-        messages.success(request, "Your account has been created succesfully!")
-        return redirect("login")
-
-    return render(request, "register.html") 
+        #messages.success(request, "Your account has been created succesfully!")
+        return JsonResponse({"status" : "successfull"})
 
```

Comment:
Please comment the way your body looks. Whaen I try the /login request with:
{
    "username": "Martinakis",
    "first_name": "m1",
    "last_name": "m2",
    "email": "m@m",
    "password1": "m3",
    "password2": "m3"
}
It returns error page and code 500 "Internal server error"
It also loggs:
django.utils.datastructures.MultiValueDictKeyError: 'username'
[12/Feb/2022 15:42:53] "POST /chats/register HTTP/1.1" 500 72636

So I guess the reques might have to be different.

Reply:
goood its fixed